UKHomeOffice/sgmr-service
# SPIKE-Pagination: Implement a version of pagination (without BE changes)

## Description
Pagination is needed on the manage voyage plans plan for when the number of voyage plans is greater than 10 (sum of drafts, cancelled and submitted). If there are already 10 voyage plans, when a user creates another, it does not appear in the UI. Pagination is needed to view all voyage plans associated with a users account.

This is a version of pagination that works without the need for the backend changes. This version will paginate all voyage plans regardless of status (even those with a 'failed' status, which will not appear in the UI). When the backend work is completed, this version of pagination will be updated to paginate according to the statuses of the voyage plans, rather than paginate them simultaneously. 

### To test

- From the dashboard, click on 'View existing voyage plans' 

**(If you have more than 10 voyage plans associated with your account)**

- You should see the option to paginate through you voyage plans

**(If you do not have more than 10)**
- Go to the code and open ManageReports.jsx
- Scroll to the pagination props and change the value of totalCount to a larger number (e.g. 108)
- 
- Click on the next button and another page should load
- Click on the previous button and the previous page should load
- Click on the numbers and you should be directed to the relevant page
- If you have 7+ pages, you should see dots (...) and siblings (pages between the dots and the current page) in the correct places
- The results (Showing 1 to 10 of 15) should also update depending on the page number

## Developer Checklist

\* Required






Ensure you delete the branch once the Pull Request is merged.
